src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -612,99 +669,159 @@ def realm_rm_zone_creds(self, realm_token_b64):
+        secret = realm_token.secret
+        try:
+            realm_info = self.realm_op().pull(EntityName(realm_token.realm_name),
+                                              realm_token.endpoint, access_key,
+                                              secret, set_default=True)
+        except RGWAMException as e:
+            raise RGWAMException('failed to pull realm', e)
+
+        realm_name = realm_info['name']
+        realm_id = realm_info['id']
+        logging.info(f"Pulled realm {realm_name} ({realm_id})")
 
-        # if endpoints:
-        #    eps = endpoints.split(',')
-        #    ep = ''
-        #    if len(eps) > 0:
-        #        ep = eps[0]
-        #        o = urlparse(ep)
-        #        port = o.port
-        #        spec = RGWSpec(service_id = svc_id,
-        #                       rgw_realm = realm_name,
-        #                       rgw_zone = zone_name,
-        #                       rgw_frontend_port = o.port)
-        #        self.env.mgr.apply_rgw(spec)
+        realm = EntityID(realm_id)
+        zonegroup = self.create_zonegroup(realm, zonegroup_name, zonegroup_is_master, endpoints)
+        self.update_period(realm, zonegroup)
 
-        self.env.mgr.apply_rgw(svc_id, realm_name, zone_name)
+        return (0, f'Created zonegroup {zonegroup_name} on realm {realm.name}', '')
 
-        daemons = self.env.mgr.list_daemons(svc_id, 'rgw', refresh=True)
+    def zone_create(self, rgw_spec, start_radosgw):
 
-        ep = []
-        for s in daemons:
-            for p in s.ports:
-                ep.append('http://%s:%d' % (s.hostname, p))
+        if not rgw_spec.rgw_realm_token:
+            raise RGWAMException('missing realm token')
+        if rgw_spec.rgw_zone is None:
+            raise RGWAMException('Zone name is a mandatory parameter')
+        if rgw_spec.rgw_zone in self.zone_op().list():
+            raise RGWAMException(f'Zone {rgw_spec.rgw_zone} already exists')
 
-        log.error('ERROR: ep=%s' % ','.join(ep))
+        realm_token = RealmToken.from_base64_str(rgw_spec.rgw_realm_token)
+        if realm_token.endpoint is None:
+            raise RGWAMException('Provided realm token has no endpoint')
 
+        access_key = realm_token.access_key
+        secret = realm_token.secret
         try:
-            zone_info = self.zone_op().modify(zone, zg, endpoints=','.join(ep))
+            realm_info = self.realm_op().pull(EntityName(realm_token.realm_name),
+                                              realm_token.endpoint, access_key, secret)
         except RGWAMException as e:
-            raise RGWAMException('failed to modify zone', e)
+            raise RGWAMException('failed to pull realm', e)
 
-        return (0, success_message, '')
+        logging.info(f"Pulled realm {realm_info['name']} ({realm_info['id']})")
+        realm_name = realm_info['name']
+        realm_id = realm_info['id']
+        logging.info(f"Pulled realm {realm_name} ({realm_id})")
```

Comment:
duplicate log statement as the one a few lines above?

Reply:
fixed